service/Microsoft.Azure.Devices/AmqpServiceClient.cs

```diff
@@ -127,15 +127,26 @@ public SendingAmqpLink SendingLink
             {
                 throw new ArgumentNullException("message");
             }
-
             Outcome outcome;
+
             using (AmqpMessage amqpMessage = message.ToAmqpMessage())
             {
                 amqpMessage.Properties.To = "/devices/" + WebUtility.UrlEncode(deviceId) + "/messages/deviceBound";
                 try
                 {
                     SendingAmqpLink sendingLink = await this.GetSendingLinkAsync();
-                    outcome = await sendingLink.SendMessageAsync(amqpMessage, IotHubConnection.GetNextDeliveryTag(ref this.sendingDeliveryTag), AmqpConstants.NullBinary, this.OperationTimeout);
+                    if (timeout != null)
+                    {
+                        outcome = await sendingLink.SendMessageAsync(amqpMessage, IotHubConnection.GetNextDeliveryTag(ref this.sendingDeliveryTag), AmqpConstants.NullBinary, (TimeSpan)timeout);                        
+                    }
+                    else
+                    {
+                        outcome = await sendingLink.SendMessageAsync(amqpMessage, IotHubConnection.GetNextDeliveryTag(ref this.sendingDeliveryTag), AmqpConstants.NullBinary, this.OperationTimeout);
```

Comment:
@ryan-z-johnson please add another test that pins down this behavior. (i.e. set `this.OperationTimeout` to something small. Calling `SendMessageAsync` should have the same behavior as calling it with a non-null timeout).

Reply:
This.OperationTimeout is not settable, so won't be able to test changing that to something small without changing the source code. I have added a test for the SendMessageAsync with a null timeout. @CIPop